### DISCUSSION TITLE
AlphaSynth as a Standalone Project for MIDI Import and Playback

### DISCUSSION THREAD

**Discussion by csa8820** [CoderLine/alphaTab#1664] [Ideas]
Currently, AlphaSynth is tightly integrated into alphaTab for rendering and playback of music notation. I would like to suggest an enhancement: making AlphaSynth a standalone project that can be used to import and play MIDI files directly, independent of alphaTab.

**Comment by Danielku15** [comment]
In the past I maintained alphaSynth independent project but it was not really worth the effort. There was not really any other user than alphaTab and it made feature development more tricky than it needed to be. Unless there is really a wide interest its unlikely that I will slice alphaSynth again into an own project. 

But if there is at least a bit of interest I could expose some a more easy-to-use standalone alphaSynth API and internals from the alphaTab package. Generally alphaSynth is quite separate within alphaTab so that should be fairly easy to do. It can be used independently (if you know how). 

Currently the main usecase for alphaSynth is to be a soundfont based midi **player**. Opening alphaSynth as own project it likely would need to serve further use cases which are not fully clear without real users (e.g. for building a web keyboard with live playing features).

If you have some concrete needs would be amazing if you can share them 😉
